app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt

```diff
@@ -159,4 +178,42 @@ class TabsTrayFragment : AppCompatDialogFragment(), TabsTrayInteractor {
             isUserInputEnabled = false
         }
     }
+
+    private fun setupNewTabButtons(currentPage: Int) {
```

Comment:
This logic looks good, but I'd recommend moving it to a separate component so that we don't make our root fragment class too big.

Since we have a `selectedPage` in the store, we can hook it up so that the value is updated when a page is selected. We can do that by dispatching an action when we observe the `TabLayout.Tab` changing [with this patch](https://github.com/jonalmeida/fenix/commit/56bcca14b68ee71e0516e311790dab015d42cddd).

With that working, you can follow the pattern of the "feature" classes in AC/Fenix that observe a property (or a set of them), and apply a change on it. Here's what the observing code would look like:

```kotlin
tabsTrayStore.flowScoped { flow ->
    flow.map { it.selectedPage }
        .ifChanged()
        .collect { page ->
            // Update FAB UI icon based on the selected page.
        }
}
```

The flow will give you the initial state when you begin observing it and all following events until your flow ends. 

Here is [a branch with a patch](https://github.com/jonalmeida/fenix/tree/fab-tabstray) based off your PR for reference. 🙂

Reply:
That's how I initially wanted to do it, but then I saw that it worked well enough in `setCurrentTrayPosition`.
But I'll change it based on your suggestion <3

Reply:
Almost forgot. 🙃  This also makes the separate class a lot easier to test without needing to mock or re-create the (Android) world for what should be a simple unit test.